### DISCUSSION TITLE
Versioning Reasoning

### DISCUSSION THREAD

**Discussion by michacassola** [ImageMagick/ImageMagick#7351] [Development] [answered]
### ImageMagick version

No specific one

### Operating system, version and so on

No specific one

### Description

I was wondering what the reasoning is behind the for me strange looking versioning with 7.X.X-X.
Isn't the last number of the 3 the one which should be used for these small increments?
So 7.1.1 to 7.1.99 instead of 7.1.1-1 to 7.1.1-99.
You must have a very good reason for this scheme as it is usually used by maintainers building a package many times with the same upstream version.

**Comment by fmw42** [comment]
For what platform did you download?  Was it a binary or from source?  Please give more details so the developers can determine how and exactly what you dowloaded and from where.

**Comment by urban-warrior** [comment, marked as answer]
Given a version number MAJOR.MINOR.MICRO-PATCH, increment the:

1. MAJOR version for incompatible API changes,
2. MINOR version when MICRO version exceeds 15 e.g. 0x70F becomes 0x710 (to accommodate PerlMagick versioning)
3. MICRO version for added functionality in backwards compatible manner, and
4. PATCH version when you make backwards compatible bug fixes.

Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.MICRO-PATCH format.
